### ISSUE TITLE
Tensor Creation from data

### ISSUE THREAD

**Issue by sashanksilwal** [microsoft/onnxruntime#13715]
### Describe the issue

Hi all,

I've exported my classificator to .onnx and have checked it in python with onnxruntime. All works perfect.
So, I am trying to run my model in c++ now but, unfortunatelly, I haven't got an enought experience in c++. I've studied examples and try to write my own code, but the moment with transform of vector to tensor is not clear for me.

Python code: 
```python 

tfidf_representation = self.sess_classification_tfidf.run(None, {"float_input": np.array([reduced_script]).reshape(1,1)})[0]
prediction = self.sess_classification.run(None, {'float_input': tfidf_representation.astype(np.float32)})[0] 
```

The shape for the tfidf model: 
<img width="357" alt="Screen Shot 2022-11-20 at 4 20 19 PM" src="https://user-images.githubusercontent.com/25387553/202901527-dd312f32-dba6-4fff-a9d2-638596352d74.png">

The shape for classification model: 
<img width="363" alt="Screen Shot 2022-11-20 at 4 21 19 PM" src="https://user-images.githubusercontent.com/25387553/202901587-38fe9f07-281b-4d44-9e75-efd269f48844.png">

How should the code be written for C++? 

My attempt:
reduced_script a string 

```c++

 const char *const input_name_classification = "float_input";
 const char *const output_name_classification = "variable";

 auto allocatorInfo = Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault);

 std::vector<int64_t> input_node_dims_classification = {1, 1}; // batch size 1, string length 1
 Ort::Value input_tensor_classification = Ort::Value::CreateTensor<float>(allocatorInfo, (float*)reduced_script.c_str(), reduced_script.size(), input_node_dims_classification.data(), input_node_dims_classification.size());

std::vector<const char*> input_names_classification = {input_name_classification};
std::vector<const char*> output_names_classification = {output_name_classification};
std::vector<Ort::Value> output_tensor_classification = sess_classification_tfidf.Run(runOptions, input_names_classification.data(), &input_tensor_classification, 499, output_names_classification.data(), 1);
        

```

Thank you very much for your help.

### To reproduce

NA

### Urgency

Urgent. The deadline for this is tomorrow. 

### Platform

Mac

### OS Version

macOS Monterey

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.13.1

### ONNX Runtime API

C++

### Architecture

X64

### Execution Provider

Default CPU

**Comment by pranavsharma** [comment]
The input type is float. Why are you casting a string to float ```(float*)reduced_script.c_str()```? Have you looked at the examples/sample code?

**Comment by sashanksilwal** [comment]
Thank you @pranavsharma 

I did look at the starter code, but I am having some issues understanding how the implementation would be for string input as opposed to images matrix. 

This is my current attempt looking at the examples, however, the size I am providing the tensor creation as output (1) and the shape of the model (499) does not match. If I pass 499 to the output_size, I get an error as the size of output_name_tfidf will be 1. 

```c++
    // create const char *const * input name float_input
    const char *const input_name_classification = "float_input";
    const char *const output_name_classification_tfidf = "variable";
    const char *const output_name_classification = "output_label";

    const char *const* input_name  = &input_name_classification;
    const char *const* output_name_tfidf = &output_name_classification_tfidf;
    const char *const* output_name = &output_name_classification;


 Ort::AllocatorWithDefaultOptions allocator;
 auto allocatorInfo = Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault);
 std::vector<int64_t> input_node_dims_classification = {1, 1};  
 Ort::Value input_tensor_classification = Ort::Value::CreateTensor<std::string>(allocatorInfo,  &reduced_script  , reduced_script.size(), input_node_dims_classification.data(), input_node_dims_classification.size());
 std::vector<Ort::Value> output_tensor_classification = sess_classification_tfidf.Run(Ort::RunOptions{nullptr}, input_name, &input_tensor_classification, 1, output_name_tfidf, 1);
 std::vector<Ort::Value> prediction = sess_classification.Run(Ort::RunOptions{nullptr}, input_name, &input_tensor_classification, 1, output_name, 1);
        
```

Here is the link to my current implementation: https://github.com/sashanksilwal/Capstone

I would be grateful if you could guide me here. Thank you

**Comment by sashanksilwal** [comment]
I was able to find the solution. I had the following errors:

1. GetShape() function returns -1 when using 'permute' and 'reshape' functions in the model.  So had to hardcode the shapes 
2. What I got after the inference was not a tensor but a sequence of map<int64, Tensor(float)>. So I could not use GetTensorMutableData() directly. 

Solutions: 
1. Hardcode the shape 
2. Example: https://github.com/microsoft/onnxruntime/blob/9e67b88c8312e124c3127c6ee0833c110a18fd5a/onnxruntime/test/shared_lib/test_nontensor_types.cc#L33

Final version of my code: https://github.com/sashanksilwal/Capstone/blob/main/Phase3/main.cpp
